comit/src/btsieve/ethereum/watch_for_event.rs

```diff
@@ -1,39 +1,45 @@
 use crate::{
     btsieve::{
-        ethereum::{poll_interval, Event, ReceiptByHash, Topic},
-        fetch_blocks_since, BlockByHash, ConnectedNetwork, LatestBlock,
+        ethereum::{poll_interval, Event, GetLogs, ReceiptByHash, TransactionByHash},
+        BlockByHash, ConnectedNetwork, LatestBlock,
     },
-    ethereum::{Block, ChainId, Hash, Input, Log, Transaction, TransactionReceipt},
+    ethereum::{Block, ChainId, Hash, Log, Transaction},
 };
 use anyhow::Result;
-use genawaiter::GeneratorState;
 use time::OffsetDateTime;
-use tracing_futures::Instrument;
 
 pub async fn watch_for_event<C>(
     connector: &C,
-    start_of_swap: OffsetDateTime,
+    _start_of_swap: OffsetDateTime,
     expected_event: Event,
 ) -> Result<(Transaction, Log)>
 where
     C: LatestBlock<Block = Block>
         + BlockByHash<Block = Block, BlockHash = Hash>
         + ReceiptByHash
-        + ConnectedNetwork<Network = ChainId>,
+        + TransactionByHash
+        + ConnectedNetwork<Network = ChainId>
+        + GetLogs,
 {
-    matching_transaction_and_log(
-        connector,
-        start_of_swap,
-        expected_event.topics.clone(),
-        |receipt| find_log_for_event_in_receipt(&expected_event, receipt),
-    )
-    .await
+    let poll_interval = poll_interval(connector).await?;
+
+    loop {
+        let logs = connector.get_logs(expected_event.clone()).await?;
+
+        if let Some(log) = find_log_for_event(&expected_event, logs) {
```

Comment:
Regarding my other comment, you are using logs here for a generic event, aren't you?

Reply:
I am not sure I understand your question. This code works for any event.

Reply:
So you are using for deployed? I thought it would not work for deployed.

Maybe I am just not familiar enough with btsieve code.

Reply:
> So you are using for deployed? I thought it would not work for deployed.

I am not. We are not emitting an event for contracts that are deployed, hence we can't use `eth_getLogs` for that. We look for the bytecode that is being deployed instead.